pkg/tfvars/gcp/gcp.go

```diff
@@ -137,3 +169,121 @@ func generateDiskEncryptionKeyLink(keyRef *machineapi.GCPEncryptionKeyReference,
 
 	return fmt.Sprintf(kmsKeyNameFmt, projectID, keyRef.KMSKey.Location, keyRef.KMSKey.KeyRing, keyRef.KMSKey.Name)
 }
+
+// getTagValuesNames returns list of tag values Names as required by terraform APIs
+// fetched using the user provided tag ParentID, Key shortname and Value shortname.
+func getTagValuesNames(cred, projID string, tags []gcp.UserTag) (map[string]string, error) {
+	if len(tags) == 0 {
+		logrus.Debugf("getTagValuesNames: user-defined tags list is empty")
+		return nil, nil
+	}
+
+	// GCP has a rate limit of 600 requests per minute, restricting
+	// to 8 requests per second.
+	limiter := newLimiter(gcpTagsRequestRateLimit, gcpTagsRequestTokenBucketSize, true)
+
+	client, err := getTagValueClient(cred)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getTagValuesNames: failed to create gcp tag value client")
+	}
+	defer client.Close()
+
+	getTagValueReq := rscmgrpb.GetNamespacedTagValueRequest{}
+	tagValues := make(map[string]string, len(tags))
+	for _, tag := range tags {
+		if err := limiter.Wait(context.TODO()); err != nil {
+			return nil, errors.Wrapf(err, "tag values name rate limit wait failed")
+		}
+
+		getTagValueReq.Name = fmt.Sprintf("%s/%s/%s", tag.ParentID, tag.Key, tag.Value)
+		tag, err := client.GetNamespacedTagValue(context.TODO(), &getTagValueReq)
+		if err != nil {
+			return nil, errors.Wrapf(err, "fetching %s tag value name failed", getTagValueReq.Name)
+		}
+		tagValues[tag.Parent] = tag.Name
+	}
+
+	bindingClient, err := getTagBindingClient(cred)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getTagValuesNames: failed to create gcp tag binding client")
+	}
+	defer bindingClient.Close()
+
+	if err := findTagConflicts(bindingClient, projID, tagValues); err != nil {
+		return nil, err
+	}
+
+	return tagValues, nil
+}
+
+// newLimiter returns token bucket based request rate limiter after initializing
+// the passed values for limit, burst(or token bucket) size. If opted for emptyBucket
+// all initial tokens are reserved for the first burst.
+func newLimiter(limit, burst int, emptyBucket bool) *rate.Limiter {
+	limiter := rate.NewLimiter(rate.Every(time.Second/time.Duration(limit)), burst)
+
+	if emptyBucket {
+		limiter.AllowN(time.Now(), burst)
+	}
+
+	return limiter
+}
+
+// getTagValueClient returns the client to be used for fetching the tag value details.
+func getTagValueClient(cred string) (*rscmgr.TagValuesClient, error) {
+	opts := []option.ClientOption{
+		option.WithCredentialsJSON([]byte(cred)),
+		option.WithEndpoint(tagRequestEndPoint),
+	}
+
+	return rscmgr.NewTagValuesRESTClient(context.TODO(), opts...)
+}
```

Comment:
It would be good to follow the patterns we have for GCP clients: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/gcp/client.go

Reply:
suggestion incorporated.